starter_code/src/movies.js

```diff
@@ -1,19 +1,98 @@
 /* eslint no-restricted-globals: 'off' */
 
-// Iteration 1: All rates average - Get the average of all rates with 2 decimals 
+// Iteration 1: All rates average - Get the average of all rates with 2 decimals
+
+function ratesAverage(array) {
+  const average =
+    array.reduce(function(accumulator, value) {
+      return accumulator + parseFloat(value.rate);
+    }, 0) / array.length;
+  return average;
+}
 
- 
 // Iteration 2: Drama movies - Get the average of Drama Movies
 
+// should I change this to some if, return null thing
+
+// average
+function dramaMoviesRate(array) {
+  let dramAve = [];
+  for (var index = 0; index < array.length; index++) {
+    if (array[index].genre.includes("Drama")) {
+      dramAve.push(array[index]);
+    }
+  }
+  const sum = dramAve.reduce((total, value) => total + value.rate, 0);
```

Comment:
One piece of advice for one of the main principles of coding - DRY (Dont repeat yourself). In this case you wrote basically the same function as you did before in the ratesAverage, so you didnt't need to rewrite it :)